src/apiv2.ts

```diff
@@ -0,0 +1,272 @@
+import fetch, { Response, RequestInit } from "node-fetch";
+
+import { FirebaseError } from "./error";
+import * as logger from "./logger";
+import * as responseToError from "./responseToError";
+import * as scopes from "./scopes";
+
+const CLI_VERSION = require("../package.json").version;
+
+const VALID_METHODS = new Set(["GET", "PUT", "POST", "DELETE", "PATCH"]);
+
+type FirebaseRequestOptions = {
+  method?: string;
+  baseURL: string;
+  path: string;
+  headers?: { [key: string]: string };
+  json?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  responseType?: "json";
+  searchParams?: { [key: string]: string | number };
+};
+
+interface FirebaseRequestHandlingOptions {
+  auth?: boolean;
+  log?: {
+    searchParams?: boolean;
+    body?: boolean;
+    resBody?: boolean;
```

Comment:
Don't these all default to `true`? Are you proposing to default them to `false`? Otherwise we should avoid having `true` by omission options.

Reply:
Maybe I misunderstood what these were for. My interpretation of these options were "these are good for debugging while coding", and could be turned on as necessary to print out more (or sensitive) information about a request.

Thinking back, is it intended to be the opposite? Is it desirable to print out as much information as possible by default rather than prevent it from being logged? (I generally agree with your thoughts though)

Reply:
By default we debug log lots of information about requests/responses so that if there are errors we have a detailed log to dig into. But in some cases the request body is very large (e.g. hosting uploads), the response body is very large, or the query params contain sensitive information (auth-based requests).

It's not meant for use while developing, it's meant for "what do we want to show up in debug logs our customers paste into GitHub"

Reply:
Got it. I've swapped this around to `skipLog`